docs(dp): Fix 1.7.0 Domain Proxy Documentation (#12615)
magma/magma: docs/docusaurus/versioned_docs/version-1.7.0/dp/debug_logs.md

---
id: version-1.7.0-debug_logs
title: Debugging and logs
hide_title: true
original_id: debug_logs
---

# Debugging

Debugging Domain Proxy usually comes down to checking log content of individual pods. This document describes how to
access these logs.

## eNB <-> AGW/enodebd

> To debug eNB side of Domain Proxy functionality, it typically falls down to inspecting the TR069 session data or

Reviewer: ```suggestion
To debug the eNB side of Domain Proxy functionality, it typically falls down to inspecting the TR069 session data or
```

Reviewer: "comes down" instead of "falls down"?

Author: `the` added.
`falls down`->`comes down`

looking into eNB specific data model/state machine implementation in `enodebd`.

> To view eNB logs from a TR069 session with `AGW`/`enodebd`, login to AGW and look for `enodebd` log file:

Reviewer: ```suggestion
To view eNB logs from a TR069 session with `AGW`/`enodebd`, login to AGW and look for the `enodebd` log file:
```

Author: Changed as suggested.

    2022-04-25 15:41:22,662 DEBUG Fetched Transient Params: {...}
    2022-04-25 15:41:22,663 DEBUG State transition from <WaitGetTransientParametersState> to <get_params>
    2022-04-25 15:41:22,664 DEBUG State transition from <GetParametersState> to <wait_get_params>
    2022-04-25 15:41:22,664 DEBUG Sending TR069 message: GetParameterValues {...}
    2022-04-25 15:41:22,671 DEBUG 10.0.2.243 - "POST / HTTP/1.1" 200 3242
    2022-04-25 15:41:22,718 DEBUG Handling TR069 message: GetParameterValuesResponse {...}
    2022-04-25 15:41:22,719 DEBUG Received CPE parameter values: {...}
    2022-04-25 15:41:22,719 DEBUG State transition from <WaitGetParametersState> to <get_obj_params>
    2022-04-25 15:41:22,720 DEBUG State transition from <GetObjectParametersState> to <wait_get_obj_params>
    2022-04-25 15:41:22,720 DEBUG Sending TR069 message: GetParameterValues {...}
    2022-04-25 15:41:22,723 DEBUG 10.0.2.243 - "POST / HTTP/1.1" 200 877
    2022-04-25 15:41:22,769 DEBUG Handling TR069 message: GetParameterValuesResponse {...}
    2022-04-25 15:41:22,770 DEBUG Received object parameters: {...}
    ```

> 1. `SetParameterValues` message followed by `GetParameterValues` - conditional, only appears if a configuration change is needed to be done on the eNB side. The actual parameters that are set in order to enable transmission based on Domain Proxy grant may differ depending on the eNB model and eNB current configuration state. Please refer to your eNB device model implementation in `lte/gateway/python/magma/enodebd/devices`.

Reviewer: ```suggestion
1. `SetParameterValues` message followed by `GetParameterValues` - conditional, only appears if a configuration change needs to be done on the eNB side. The actual parameters that are set in order to enable transmission based on Domain Proxy grant may differ depending on the eNB model and eNB current configuration state. Please refer to your eNB device model implementation in `lte/gateway/python/magma/enodebd/devices`.
```

Author: Changed as suggested.

    ```console
    2022-04-25 15:41:22,771 DEBUG State transition from <WaitGetObjectParametersState> to <set_params>`
    2022-04-25 15:41:22,772 DEBUG Sending TR069 request to set CPE parameter values: {...}`
    2022-04-25 15:41:22,773 DEBUG State transition from <SetParameterValuesState> to <wait_set_params>`
    2022-04-25 15:41:22,773 DEBUG Sending TR069 message: SetParameterValues {...}`
    2022-04-25 15:41:22,777 DEBUG 10.0.2.243 - "POST / HTTP/1.1" 200 723`
    2022-04-25 15:41:22,837 DEBUG Handling TR069 message: SetParameterValuesResponse {'Status': '0'}`
    2022-04-25 15:41:22,838 INFO Successfully configured CPE parameters!`
    2022-04-25 15:41:22,839 DEBUG State transition from <WaitSetParameterValuesState> to <check_get_params>`
    2022-04-25 15:41:22,840 DEBUG State transition from <GetParametersState> to <check_wait_get_params>`
    2022-04-25 15:41:22,840 DEBUG Sending TR069 message: GetParameterValues {...}`
    2022-04-25 15:41:22,849 DEBUG 10.0.2.243 - "POST / HTTP/1.1" 200 3242`
    2022-04-25 15:41:22,916 DEBUG Handling TR069 message: GetParameterValuesResponse {...}`
    ```

> 1. Update transmission configuration from Domain Proxy and close TR069 session - `notify_dp` is a `enodebd` transition state that calls Domain Proxy `GetCBSDState` gRPC API. `GetCBSDState` response contains data, which indicate whether eNB radio should be disabled/turned off or enabled together with transmission parameters. `GetCBSDState` response data is translated to eNB specific parameters that will be applied to eNB. Please refer to your eNB device model implementation in `lte/gateway/python/magma/enodebd/devices`.

Reviewer: ```suggestion
1. Update transmission configuration from Domain Proxy and close TR069 session - `notify_dp` is an `enodebd` transition state that calls the Domain Proxy `GetCBSDState` gRPC API. The `GetCBSDState` response contains data, which indicate whether eNB radio should be disabled/turned off or enabled together with transmission parameters. `GetCBSDState` response data is translated to eNB specific parameters that will be applied to eNB. Please refer to your eNB device model implementation in `lte/gateway/python/magma/enodebd/devices`.
```

Author: Changed as suggested.

Reviewer: Looks like you accidentally kept the "a" when making the change, as it now reads "a an"

Author: Oh, good spot. Fixed.

Apr 21 11:57:55 FreedomFi-Gateway control_proxy[1220289]: 2022-04-21T11:57:54.991Z [127.0.0.1 -> dp_service-controller.codistaging.dp.freedomfi.com,8443] "POST /DPService/GetCBSDState
Apr 21 11:57:50 FreedomFi-Gateway control_proxy[1220289]: 2022-04-21T11:57:50.044Z [127.0.0.1 -> dp_service-controller.codistaging.dp.freedomfi.com,8443] "POST /DPService/GetCBSDState
Apr 21 11:57:45 FreedomFi-Gateway control_proxy[1220289]: 2022-04-21T11:57:45.002Z [127.0.0.1 -> dp_service-controller.codistaging.dp.freedomfi.com,8443] "POST /DPService/GetCBSDState
```

`control_proxy` logs will show if gRPC calls from `enodebd` towards Domain Proxy are made. Since gRPC is binary coded, there isn't much information from tcpdump capture on this end of communication.
Domain Proxy gRPC call details can be viewed in [NMS](#nms).

### NMS

`enodebd` gRPC calls towards Domain Proxy are visible in NMS: `[Metrics]` menu, `[DP Logs]` tab in a human readable form.
An `empty` message (blank content in the column) is equivalent to no SAS grant data being sent - which in turn is interpreted by `AGW`/`enodebd` as

> no transmission and the radio transmit must be disabled.

Reviewer: "transmission" instead of "transmit"?

Author: Changed as suggested.

## Check individual pods' logs

> To view logs from individual Domain Proxy pods, execute `kubectl logs` command with one of the pod names listed in [previous chapter].(#listing-domain-proxy-pods-in-kubernetes)

Reviewer: ```suggestion
To view logs from individual Domain Proxy pods, execute `kubectl logs` with one of the pod names listed in the [previous chapter](#listing-domain-proxy-pods-in-kubernetes)
```

Author: Changed as suggested.

- `Radio Controller` (RC) logs:
    - logs related to AGW/enodebd <-> Domain Proxy communication
    - logs related to requests generated by Active Mode Controller (Domain Proxy logic, which generates appropriate SAS requests)

>     - logs related with Database modifications, which where the result of incoming API calls (either from AGW or AMC)

Reviewer: ```suggestion
    - logs related with Database modifications, which were the result of incoming API calls (either from AGW or AMC)
```

Author: Changed as suggested.
